src/components/Navbar/index.tsx

```diff
@@ -1,20 +1,124 @@
+        const { value } = event.target
+        setFormData(value)
+        console.log(value);
+        const filter = data.filter((item) => item.title.toLowerCase().includes(value.toLowerCase()))
+        setResult(filter)
+    }
+    // Busqueda por titulo
+    const handleSearch = (event: any) => {
+        event.preventDefault()
+        const filter = data.filter((item) => item.title.toLowerCase().includes(formData.toLowerCase()))
+        setResult(filter)
+    }
+
+    const fetchData = async (endpoint: any) => {
+
+        try {
+            const response = await fetch(endpoint, {
+                method: 'GET',
+                headers: {
+                    'Content-Type': 'application/x-www-form-urlencoded',
+                },
+            });
+
+            if (!response.ok) {
+                throw new Error(`Error: ${response.status} - ${response.statusText}`);
+            }
+
+            return await response.json();
+        } catch (error) {
+            console.error('Error in fetchData:', error);
+            throw error;
+        }
+    }
+
+    //Comentar este useEffect para usar mock dataJson
+   
+    // useEffect(() => {
+    //     const getData = async () => {
+    //         const res = await fetchData(endpoint)
+    //         setData(res)
+    //         setResult(res)
+    //     }
+
+    //     getData()
+
+    // }, [])
+
     return (
-        <nav className={styles.nav}>
-        <ul>
-            <li>
-                <Link href="/">Home</Link>
-            </li>
-            <li>
-                <Link href="/post">Posts</Link>
-            </li>
-            <li>
-                <Link href="/about">About</Link>
-            </li>
-        </ul>
-        </nav>
+        <>
+            <nav className={styles.nav}>
+                <ul>
+                    <li>
+                        <Link href="/">Home</Link>
+                    </li>
+                    <li>
+                        <Link href="/post">Posts</Link>
+                    </li>
+                    <li>
+                        <Link href="/about">About</Link>
+                    </li>
+                </ul>
+                <button onClick={handleClick}>
+                    <img src={SearchIcon.src} alt="searchIcon" />
+                </button>
+            </nav>
+            <div className={styles.form}>
+                <form className={`${styles.container} ${toogle ? styles.expanded : ''}`} onSubmit={handleSearch}>
+                    <input name='search' value={formData} onChange={handleChange} type="text" />
+                    <button type='submit' onClick={handleSearch}>buscar</button>
+                </form>
+                {
+                    toogle ? (
+                        <div className={styles.sectionCards}>
+                            {
+                                result.map((item, index) => (
+                                    <div key={index} className={styles.card}>
```

Comment:
Create small components. you can create another component with the name Cards or ItemCards, the best way to scale an application is by applying the separation of code.